x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java

```diff
@@ -113,7 +136,12 @@ public List<String> chunk(String input, int maxNumberWordsPerChunk) {
                     chunkWordCount = sentenceSplits.get(i).wordCount();
```

Comment:
Personally, I would move this code into `splitLongSentence` and have it return last chunk start and word count.